projects/graphql-client/src/graphql-request.service.ts

```diff
@@ -85,6 +85,11 @@ export class GraphQlRequestService {
     }
   }
 
+  // Needs to be called when some service calls error out and we want to refetch data
+  public clearCache(): void {
+    this.apollo.client.clearStore();
```

Comment:
Looks like our end goal is to refetch queries. Wouldn't it be better to use [client.refetchQueries](https://www.apollographql.com/docs/react/data/refetching/#clientrefetchqueries)?

Reply:
I and Anirban did discuss this, but this looks an overhead to refetch all the queries that we might not even need. That might in turn cause delay in the responses as it may queue up those queries on the backend, hence went with `clearStore` here.

Reply:
Also, jFYI, the relevant documentation for Angular is [here](https://the-guild.dev/graphql/apollo-angular/docs/caching/advanced-topics#resetting-the-store)